Automattic/wp-calypso
# composite-checkout: Clear cart when purchase is complete

#### Changes proposed in this Pull Request

When a purchase completes in composite checkout, the transaction endpoint on the server will empty the cart, but if we're immediately redirected to a page in calypso which displays a cart notification (eg: the Plans page), it will still think there's products in the cart since the server can't notify us of the change. 

![Screen Shot 2020-05-15 at 3 50 53 PM](https://user-images.githubusercontent.com/2036909/82090612-e734ae80-96c3-11ea-9179-46f07c60ed74.png)

The proper solution to this is to stop using the CartStore and instead query the cart when it's needed. Eventually the CartStore can be deprecated and this problem will go away, but until then we need to help out.

This PR causes the CartStore to empty the cart as soon as the transaction is complete in composite checkout. That way it should become in sync with the cart on the server.

![Screen Shot 2020-05-15 at 3 51 03 PM](https://user-images.githubusercontent.com/2036909/82090621-eac83580-96c3-11ea-8ef0-c617fe58a961.png)


#### Testing instructions

- Add a product to your cart and visit composite checkout.
- Complete a purchase using any payment method other than PayPal (PayPal is a redirect method and therefore never notifies composite checkout that it has completed; this fix will not help PayPal).
- Immediately visit the plans page and verify that the cart icon in the upper-right displays nothing in the cart.
